sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryHelpers.java

```diff
@@ -117,6 +117,14 @@ public static String stripPartitionDecorator(String tableSpec) {
   }
 
   static String jobToPrettyString(@Nullable Job job) throws IOException {
+    if (job != null && job.getConfiguration().getLoad() != null) {
+      // Removing schema and sourceUris from error messages for load jobs since these fields can be
+      // quite long and error message might not be displayed properly in runner specific logs.
+      job = job.clone();
```

Comment:
Do you know how expensive this operation is? If this gets called in a crash-loop the cost could add up quick.

If it is expensive, we could consider caching the `job.toPrettyString()` output, assuming the Job object is immutable.

But, if performance isn't a concern, then better to keep the code simple and leave as-is.

Reply:
Effect on performance seems to be insignificant. For my example job time to clone this object was < 1ms.